src/manager.cpp

```diff
@@ -52,6 +52,15 @@ sdbusplus::async::task<> Manager::init()
     {
         _ctx.spawn(monitorServiceNotifications());
     }
+
+    /**
+     * The RBMC manager is responsible for triggering both background and
+     * full sync operations once redundancy is enabled after a failover,
+     * which may change the BMC role. It should monitor the relevant RBMC
+     * manager properties and update the cached data-sync state whenever the
+     * role changes, ensuring data is synchronized according to the new role.
+     */
+    _ctx.spawn(_extDataIfaces->watchRedundancyMgrProps());
```

Comment:
Since you have a mock for it it wouldn't need to be in the non unit test only section?  I don't mind either way.

Reply:
Yeah, but the issue is that the manager object starts a “watching forever” task, so we need a way to forcefully stop this task in all test cases that create the manager. I want to avoid adding a lot of extra code across the affected unit tests, so I was considering ignoring the watch operation entirely in the UT context.

Since we’re planning to define a sync-specific sender/receiver framework with proper cancellation support, I’d like to address the current UT challenges as part of that work.